### PR TITLE
Add verifiedOnBlockchain property  to transaction meta

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation

This PR aims to add `verifiedOnBlockchain` property to transaction metadata as part of the ongoing effort to align the extension transaction controller with the controller in the core repo.

For context, the core `TransactionController` uses a `verifiedOnBlockchain` property to indicate whether a transaction has completed the verification process and therefore the status is now fixed and no further checks will be performed.

If the transaction receipt indicates the transaction failed, the `verifiedOnBlockchain` is still set to `true` to prevent further processing.

**Changes**
- added `verifiedOnBlockchain` to transaction metadata.
- Created migration to set the value accordingly on existing transactions in the controller state.
- added a check in `PendingTransactionTracker`.

## Manual Testing Steps

**Scenario 1**
- Go to the test-dapp
- Create a token
- Perform a transaction

**Scenario 2**
- Switch to a test net
- Send eth

Both scenarios should be successfully completed, as usual, with no functional changes.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by sleepytanya** [comment]
Both scenarios work as expected.
1. Sending Sepolia ETH:

https://github.com/MetaMask/metamask-extension/assets/104780023/a8e8bc17-c8eb-4823-8f23-3e1b5575c188
Sending tokens:



https://github.com/MetaMask/metamask-extension/assets/104780023/9d7fb457-fd01-4e7c-827e-02bb826cf22d

File: app/scripts/controllers/transactions/index.js [hunk 1]
```diff
@@ -963,6 +963,8 @@ export default class TransactionController extends EventEmitter {
       if (blockTimestamp) {
         txMeta.blockTimestamp = blockTimestamp;
       }
+      // transaction verified on the blockchain
```
Comment (matthewwalsh0) [inline app/scripts/controllers/transactions/index.js:966]: Minor, but is this comment adding much additional detail?

File: app/scripts/controllers/transactions/pending-tx-tracker.js [hunk 2]
```diff
@@ -185,6 +185,11 @@ export default class PendingTransactionTracker extends EventEmitter {
       return;
     }
 
+    // Skip processing transactions verified on the blockchain
+    if (txMeta.verifiedOnBlockchain) {
```
Comment (matthewwalsh0) [inline app/scripts/controllers/transactions/pending-tx-tracker.js:189]: Should we join this with the condition above for more compact code?

File: app/scripts/controllers/transactions/pending-tx-tracker.js [hunk 3]
```diff
@@ -185,6 +185,11 @@ export default class PendingTransactionTracker extends EventEmitter {
       return;
     }
 
+    // Skip processing transactions verified on the blockchain
```
Comment (matthewwalsh0) [inline app/scripts/controllers/transactions/pending-tx-tracker.js:188]: Minor, but is the name of the property already explicit enough here?

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**